python/paddle/distributed/auto_parallel/operators/dist_embedding.py

```diff
@@ -102,118 +102,107 @@ def update_dims_mapping(self, op_dist_attr):
-            model_parallel_axis, process_mesh = op_dist_attr.get_owner_context(
-            )._get_model_parallel_info()
-            group_ranks = _get_comm_group(process_mesh.process_group,
-                                          process_mesh.topology,
-                                          model_parallel_axis, rank_id)
-            group = new_process_group(group_ranks)
-
-            # append op
-            check_variable_and_dtype(Ids_var, 'input', ['int32', 'int64'],
-                                     'c_embedding')
-
-            intermediate_var_0 = dst_block.create_var(
-                name=unique_name.generate_with_ignorable_key(".".join(
-                    ["c_embedding", 'tmp'])),
-                dtype=Weight_var.dtype,
-                shape=Out_var.shape,
-                type=core.VarDesc.VarType.LOD_TENSOR,
-                persistable=False,
-                stop_gradient=Out_var.stop_gradient)
-            # copy Out_var's dist_attr to intermediate_var_0's dist_attr
-            copy_distributed_attr_for_var(op_dist_attr, intermediate_var_0,
-                                          Out_var)
-
-            check_variable_and_dtype(
-                Out_var, 'tensor',
-                ['float16', 'float32', 'float64', 'int32', 'int64'],
-                'c_allreduce_sum')
-
-            c_embedding_op = dst_block.append_op(
-                type='c_embedding',
-                inputs={'Ids': [Ids_var],
-                        'W': [Weight_var]},
-                outputs={'Out': [intermediate_var_0]},
-                attrs={"start_index": relative_idx})
-
-            # use_model_parallel
-            c_allreduce_sum_op = dst_block.append_op(
-                type='c_allreduce_sum',
-                inputs={'X': [intermediate_var_0]},
-                outputs={'Out': [Out_var]},
-                attrs={
-                    'ring_id': group.id,
-                    'use_calc_stream': True,
-                    'use_model_parallel': True,
-                })
-
-            # copy serial op's dist_attr to dist op's dist_attr
-            copy_distributed_attr_for_dist_op(c_embedding_op, dst_block,
-                                              op_dist_attr)
-            copy_distributed_attr_for_dist_op(c_allreduce_sum_op, dst_block,
-                                              op_dist_attr)
-
-        if in_dygraph_mode():
-            raise NotImplementedError(
-                "Dist op for [{}] with idx [{}] is NOT implemented yet.".format(
-                    "matmul", 0))
+    @staticmethod
+    def forward(ctx, *args, **kwargs):
+        """
+        kwargs: inputname_mapping & outputname_mapping
+        """
+        dst_block = ctx.get_dst_main_program().global_block()
+        src_op = ctx.get_cur_src_op()
+        rank_id = ctx.get_rank_id()
+        op_dist_attr = ctx.get_cur_dist_attr()
+
+        # check validation of inputs / outputs 
+        for input_name in src_op.desc.input_names():
```

Comment:
for a specific dist op(e.g. matmul) we already know that we need "X" and "Y" and "Out" in its intput/output, it would be better to name that in the code.